icu4c/source/test/cintltst/utf16tst.c

```diff
@@ -397,6 +397,10 @@ static void TestNextPrevChar(void){
 
 }
 
+#if U_GCC_MAJOR_MINOR >= 406
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds"
```

Comment:
Please add a comment to explain why the code is OK and the compiler is mistaken in warning about it.

Reply:
@markusicu Could you look at this test? This test is written and submitted by you in https://github.com/unicode-org/icu/commit/edac6e7206ee832a05d54b62da2a5e09ead3f907
From what I can tell, the legnth cannot be -1 for these macro, based on the macro description. 
```
 * @param i string offset, must be i<length
 * @param length string length
```

but you pass in -1 for length when calling   U16_NEXT(), U16_NEXT_OR_FFFD(),  U16_FWD_1(),  U16_FWD_N(), U16_GET(),  U16_SET_CP_LIMIT() 

I think that is one of the reason casuing the warning.